src/lib/grub.rs

```diff
@@ -142,8 +178,8 @@ mod tests {
     #[test]
     fn test_boot_counter_set() {
         let (_temp_dir, grubenv) = setup_test_paths();
-        set_boot_counter(10, &grubenv, MOUNT_INFO_PATH).unwrap();
-        assert_eq!(get_boot_counter(&grubenv).unwrap(), Some(10));
+        set_boot_counter_at(10, &grubenv).unwrap();
```

Comment:
Shouldn't we test the public function `set_boot_counter` instead of the private implementation? If we test the private implementation we can reach a situation were the public function is broken but the tests pass.

Isn't it possible to modify the global `GRUB_PATH` variable to a temporary file and then test the public function? Moreover, if that's possible maybe all the _at function are not needed. The same applies to the mount library